agent/engine/task_manager.go

```diff
@@ -148,6 +149,18 @@ func (mtask *managedTask) overseeTask() {
 		if !mtask.GetKnownStatus().Terminal() {
 			// If we aren't terminal and we aren't steady state, we should be able to move some containers along
 			llog.Debug("Task not steady state or terminal; progressing it")
+
+			// TODO: Add new task resources provisioned state ?
+			err := mtask.SetupPlatformResources()
+			if err != nil {
```

Comment:
@petderek I'd vouch for getting rid of the changes to the task_manager now and following up  the changes to the task_manager once we define `platformResources` in a separate PR.